src/python/pants/vcs/git.py

```diff
@@ -106,6 +113,28 @@ def changed_files(
         # git will report changed files relative to the worktree: re-relativize to relative_to
         return {self._fix_git_relative_path(f, relative_to) for f in files}
 
+    def changed_files_lines(
+        self,
+        paths: Iterable[str],
+        from_commit: str,
+        relative_to: PurePath | str | None = None,
+    ) -> FrozenDict[str, tuple[Hunk, ...]]:
+        relative_to = PurePath(relative_to) if relative_to is not None else self.worktree
+
+        return self._diff_parser.parse_unified_diff(
+            self._git_binary._invoke_unsandboxed(
+                self._create_git_cmdline(
+                    [
+                        "diff",
+                        "--unified=0",
+                        from_commit + "...HEAD",
```

Comment:
Looks like this will only find diffs from the `from_commit` to HEAD, but will ignore unstaged and uncommitted changes?

Reply:
You're right, I've added the commit diff as a POC and skipped unstaged changes. It probably makes sense to add it in this pr, I'll do it